src/hotspot/share/utilities/vmError.cpp

```diff
@@ -1312,6 +1313,27 @@ int VMError::prepare_log_file(const char* pattern, const char* default_pattern,
   return fd;
 }
 
+// Attempt to set the current thread to Native state in VMError.
+// By setting it to Native, it allows os::fork_and_exec to execute cmd such as jcmd %p.
+// Otherwise, it may end up with a deadlock when dcmds try to synchronize all Java threads
+// at safepoints.
+bool transition_into_native() {
```

Comment:
Since this is a highly specific workaround for one specific problem (forking off jcmd which will attach back to us with a sub command which needs a safe point) I would suggest naming this function somewhat more specific, e.g. "prepare_jcmd_workaround" or "prepare_fork_and_exec" (though the latter is misleading since this is needed only for the jcmd-attaches-to-me case).

For these cases, I also prefer the JBS issue number mentioned in a comment. Its helpful when one looks at the code, and one does not have to dig through the change history.

Reply:
I don't like workaround.  I intend to do thing right.  In my understanding, fork_and_exec() is going to step into Native, isn't it?

I would like to clean up [prepare_for_emergency_dump](https://github.com/openjdk/jdk/blob/master/src/hotspot/share/jfr/recorder/repository/jfrEmergencyDump.cpp#L426) in next step. It's easier to maintain than just iterating locks manually.  that's why I would like to have a natural name.  How about `VMError::transition_java_current_to_native_from_vm()`?